api/v1alpha1/vpcendpoint_types.go

```diff
@@ -77,6 +81,23 @@ type VpcEndpointStatus struct {
 
 	// Whether the Route53 CNAME record has been created
 	CNAMERecordCreated bool `json:"hostedZoneRecordCreated,omitempty"`
+
+	// ExternalServiceNameStatus is the status of the ExternalName service
+	ExternalServiceNameStatus metav1.Status `json:"externalNameServiceStatus,omitempty"`
+}
+
+type ExternalNameServiceSpec struct {
+	//Name  is the name of the externalName service
+	Name string `json:"name"`
+
+	// Namespace is the namespace of the externalName service
+	Namespace string `json:"namespace"`
+
+	// type is the type of the Kubernetes Service supporting the VPC Endpoint Service
+	ServiceType corev1.ServiceType `json:"type"`
+
+	// ExternalName is the DNS record of the Kubernetes Service supporting the VPC Endpoint Service
+	ExternalName string `json:"externalName,omitempty"`
```

Comment:
the ExternalName type should be removed since we're filling it in via

`ExternalName: fmt.Sprintf("%s.%s", resource.Spec.ServiceName, r.clusterInfo.domainName)`